wandb/sdk/wandb_run.py

```diff
@@ -2957,6 +2957,37 @@ def finish_artifact(
             finalize=True,
         )
 
+    def _download_artifact(
```

Comment:
Nit: `Run` should not handle artifacts logic, let's move this to the `Artifact` class.

Reply:
How will the artifact communicate with the backend interface in that case though? I tried instantiating a backend object in the `download()` method of the Artifact class, via `backend = Backend()` but that creates a circular import. 

Did you have something specific in mind?

Reply:
Is this what you meant? https://github.com/wandb/wandb/blob/cb77649b44991cf5ed7f6cb29672625ff6b3bee8/wandb/sdk/artifacts/artifact.py#L1689

Reply:
We can get the backend instance from the run, but the logic can be in `Artifact`.